### PR TITLE
Return best distance

### LINKED ISSUE #2: return best distance
it would be super useful in many cases to know what the best distance was, for example to decide whether or not to display a label, or to use in other computations.

Non-breaking change 1:
`var r = [bestCell.x, bestCell.y]; r.distance = bestCell.d; return r;`

Not-too-much-breaking change 2:
`return [bestCell.x, bestCell.y, bestCell.d];`

1 is probably nicer but some developers might find the array/object mix awkward to work with.

### PR DESCRIPTION
**Pull request by anandthakker**
Closes #2 

@mourner dunno if you actually want this in, but I needed it for something so here ya go

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -69,7 +69,9 @@ function polylabel(polygon, precision, debug) {
         console.log('best distance: ' + bestCell.d);
     }
 
-    return [bestCell.x, bestCell.y];
+    var result = [bestCell.x, bestCell.y];
+    result.distance = bestCell.d;
```
Comment (jingsam) [inline index.js]: An array mixed with object seems wired

**Comment by TWiStErRob** [comment]
Why not swallow the "breaking change" and bump the version number accordingly?
```
return {x: bestCell.x, y: bestCell.y, distance: bestCell.d};
```

**Comment by anandthakker** [comment]
I have no personal opinion on this question -- attaching `distance` to an array, switching to an object, switching to a 3-element array all seem fine to me.  I'll leave it to @mourner to state a preference.

File: index.js (hunk 1 above)
Comment (curran) [inline index.js]: I've seen properties on arrays in D3, for example `data.columns` in [dsv.parse](https://github.com/d3/d3-dsv#dsv_parse)

**Comment by curran** [comment]
This would be a cool feature to have. I'd say the property on the array is fine.

**Comment by anandthakker** [comment]
Closing this as stale.

**Comment by TWiStErRob** [comment]
So what about #2? The fix is here, a button press away; someone just needs to own the decision of what format to use.

**Comment by jolars** [comment]
I would love to see this implemented so that there might be a solution for multiPolygons.

**Comment by mourner** [comment]
Superceded by #61

**Pull request closed without merging**